eth2near/eth2near-block-relay-rs/src/execution_block_proof.rs

```diff
@@ -90,19 +93,21 @@ impl ExecutionBlockProof {
             l2_proof,
             Self::L2_EXECUTION_PAYLOAD_PROOF_SIZE,
             Self::L2_EXECUTION_PAYLOAD_TREE_EXECUTION_BLOCK_INDEX,
-        );
+        )?;
 
-        merkle_proof::verify_merkle_proof(
+        Ok(merkle_proof::verify_merkle_proof(
             execution_payload_hash,
             l1_proof,
             BeaconBlockBodyMerkleTree::BEACON_BLOCK_BODY_TREE_DEPTH,
             Self::L1_BEACON_BLOCK_BODY_TREE_EXECUTION_PAYLOAD_INDEX,
             *beacon_block_body_hash,
-        )
+        ))
     }
 
-    fn merkle_root_from_branch(leaf: H256, branch: &[H256], depth: usize, index: usize) -> H256 {
-        assert_eq!(branch.len(), depth, "proof length should equal depth");
+    fn merkle_root_from_branch(leaf: H256, branch: &[H256], depth: usize, index: usize) -> Result<H256, Box<dyn Error>> {
+        if branch.len() != depth {
+            return Err(Box::new(IncorrectBranchLength));
+        }
```

Comment:
I think we need to avoid boxing errors, it seems to be redundant in this case (and in the calling function too).

Reply:
fix https://github.com/aurora-is-near/rainbow-bridge/pull/823/commits/3edebc4e1a26fc0cf705760f78b01c69f1756ffb